extmod/network_cyw43.c

```diff
@@ -233,6 +241,10 @@ STATIC mp_obj_t network_cyw43_connect(size_t n_args, const mp_obj_t *pos_args, m
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
+    #ifdef hal_check_network_allowed
+    hal_check_network_allowed(MP_QSTR_WLAN);
+    #endif
```

Comment:
I don't think this is needed, because the WLAN must be active before you can call `.connect()`.

Reply:
Not exactly true. It only has to be active the first time. The following works. Maybe this is a bug that should be fixed (in the driver)?

import network
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.active(False)
wlan.connect('Pi Towers Guest', 'welcometopitowers') #<-- SUCCEEDS

Reply:
That's a bug...

But at the same time, it'd probably be nice if `wlan.connect()` automatically activated the interface.  Well, that's out of scope for what we are doing here.

Reply:
See other comment - I disagree, wlan.active(True) just loads the driver. For STA mode it's connnect/disconnect that matters. I've tried to clear up the behaviour of the API with the latest change.